### ISSUE TITLE
Test zypper integration in pulp_rpm

### ISSUE THREAD

**Issue by sbernhard** [PulpQE/pulp-smash#674]
Currently, pulp_rpm is using the yum backend to install/uninstall/update packages. There are other backends like dnf and zypper. 

As SUSE doesn't use yum but zypper, pulp_rpm does use the distribution flag to choose the zypper backend. 

For testing pulp_rpm you need to have access to a OpenSUSE / SUSE distribution and install pulp_rpm. Installing / uninstalling / updating software should then be done by using zypper instead of yum.

This testing request pairs with this code: https://github.com/pulp/pulp_rpm/pull/1041  I don't think there is an issue filed for this.

**Comment by Ichimonji10** [comment]
To the future implementer: there are two steps to solving this issue. The first step is to update [`pulp_smash.cli`](https://github.com/PulpQE/pulp-smash/blob/master/pulp_smash/cli.py). That module defines methods like `PackageManager.install` that call out to dnf or yum, depending on which is available. That module should be expanded so that it takes zypper into account. The second step is to update tests that muck around in `/etc/yum.repos.d/` as needed, such as [`pulp_smash.tests.rpm.cli.test_copy_units`](https://github.com/PulpQE/pulp-smash/blob/master/pulp_smash/tests/rpm/cli/test_copy_units.py). The canonical way of adding a repository on a zypper-based system seems to be through zypper itself, such as with `zypper addrepo ...`. The tests that muck around in `/etc/yum.repos.d` should probably be updated to add repos in a more generic way.

**Comment by bmbouter** [comment]
@sbernhard So the commands to test are install, uninstall, and update? What about binding and unbinding a consumer from a repo? Is that support?

Also it sounds like it should "just work" if the client is OpenSUS / SUSE. Is that right?

**Comment by sbernhard** [comment]
Hi, 

I have developed the zypper plugin using foreman / katello. Therfore I don't know about binding and unbinding - I'm sorry. 

Zypper is a suse tool and I don't know if it's available somewhere else. Therefore I added a check if the system is suse /openSUSE. 

Best regards 
Bernhard Suttner 

---- bmbouter schrieb ----

@sbernhard So the commands to test are install, uninstall, and update? What about binding and unbinding a consumer from a repo? Is that support?

Also it sounds like it should "just work" if the client is OpenSUS / SUSE. Is that right?
